pac4j-core/src/main/java/org/pac4j/core/credentials/authenticator/LocalCachingAuthenticator.java

```diff
@@ -56,8 +59,30 @@ public void validate(final T credentials, final WebContext context) throws HttpA
             });
             credentials.setUserProfile(profile);
             logger.debug("Found cached credential. Using cached profile {}...", profile);
-        } catch (Exception e) {
-            throw new CredentialsException(e);
+        } catch (UncheckedExecutionException e) {
+            if (e.getCause() instanceof RuntimeException) {
+                throw (RuntimeException) e.getCause();
+            } else if (e.getCause() != null) {
```

Comment:
To clarify, it's a matter of consistency and keeping the codebase lean. It would pose a problem longterm if we ended up repeating things, repeaing things repeaing things, that makes it harder to read the code read the code read the code read the code :)